dev/com.ibm.ws.microprofile.reactive.messaging.kafka/src/com/ibm/ws/microprofile/reactive/messaging/kafka/KafkaInput.java

```diff
@@ -209,9 +209,11 @@ public void shutdown() {
         if ((records != null) && !records.isEmpty()) {
             result.complete(fromIterable(records));
         } else {
-            this.executor.submit(() -> {
-                executePollActions(result);
-            });
+            if (this.running && !this.executor.isShutdown()) {
+                this.executor.submit(() -> {
+                    executePollActions(result);
+                });
+            }
```

Comment:
I would suggest a few changes here:
1. Catch the exception rather than checking `executor.isShutdown()` to eliminate the small window between that call and the call to `submit` where the executor could stop accepting new tasks.
2. Don't worry about checking `this.running`, that's already checked all over the place
3. Assume that a `RejectedExecutionException` means that the application is shutting down and treat it in the same way as the case on line 185 (trace and return an incomplete future because we're shutting down)

Reply:
done